pydicom/tests/test_dataelem.py

```diff
@@ -43,6 +45,20 @@ def testVM2(self):
         self.assertEqual(VM, 1,
                          "Wrong Value Multiplicity, expected 1, got %i" % VM)
 
+    def testDSFloatConversion(self):
+        """Test that strings are correctly converted if changing the value."""
+        self.assertTrue(isinstance(self.data_elementDS.value, DSfloat))
+        self.assertTrue(isinstance(self.data_elementMulti.value[0], DSfloat))
+        # multi-value append/insert
+        self.data_elementMulti.value.append('42.4')
+        self.assertTrue(isinstance(self.data_elementMulti.value[3], DSfloat))
+        self.data_elementMulti.value.insert(0, '42.0')
+        self.assertTrue(isinstance(self.data_elementMulti.value[0], DSfloat))
+        # change single value of multi-value
+        self.data_elementMulti.value[3] = '123.4'
+        self.assertTrue(isinstance(self.data_elementMulti.value[3], DSfloat))
+        self.assertEqual(DSfloat('123.4'), self.data_elementMulti.value[3])
```

Comment:
A minor comment: these tests could be a bit stronger if they checked that the correct value is in the correct place, rather than just the type, as the last block does for '123.4'.